File: snail-race-bet/typescript/src/domain/BetRepositoryFake.ts
import {Bet, BetRepository} from "./BetRepository";

export class BetRepositoryFake implements BetRepository {
    private bets = new Array<Bet>()
    async register(bet: Bet): Promise<void> {
        this.bets.push(bet);
    }

    async findByDateRange(from: number, to: number): Promise<Bet[]> {
        return this.bets
            .filter(b => b.timestamp >= from && b.timestamp < to)
    }
}

File: snail-race-bet/java/src/main/java/snail/race/kata/domain/BetRepositoryFake.java
package snail.race.kata.domain;

import java.util.ArrayList;
import java.util.List;

public class BetRepositoryFake implements BetRepository {
    private List<Bet> inMemoryBets = new ArrayList<>();

    @Override
    public void register(Bet bet) {
       inMemoryBets.add(bet);
    }

    @Override
    public List<Bet> findByDateRange(int from, int to) {
        return inMemoryBets.stream().filter(bet -> bet.timestamp() >= from && bet.timestamp() < to).toList();
    }
}
